File: ABCcontest/ABC451/reE.py
N=int(input())

G=[]
for i in range(1,N):
    connect=list(map(int,input().split()))
    G.append(connect)

can_make=True

for i in range(N-2):
    if not can_make:
        break
    Gi=G[i]
    for j in range(i+1,N-1):
        if not can_make:
            break
        Gj=G[j]
        t1=Gi[j-i-1]
        for k in range(j+1,N):
            t2=Gj[k-j-1]
            t3=Gi[k-i-1]
            
            if t1+t2<t3 or t1+t3<t2 or t2+t3<t1:
                can_make=False
                break

if can_make:
    print("Yes")
else:
    print("No")

File: ABCcontest/ABC451/reE.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
#include <queue>
#include <utility>
#include <map>
#include <set>
#include <cmath>

using namespace std;
using ll = long long;

#define rep(i, n) for (int i = 0; i < (int)(n); ++i)
#define all(v) (v).begin(), (v).end()


int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int N;
    cin>>N;

    vector<vector<int>> G(N,vector<int>(N));

    rep(i,N-1){
        rep(j,N-i-1){
            int num;
            cin >>num;
            G[i][j]=num;
        }

    }

    bool can_make=true;

    rep(i,N-2){
        if (not can_make){
            break;
        }
        vector<int> Gi=G[i];
        rep(j,N-i-1){
            if (not can_make){
                break;
            }
            
            vector<int> Gj=G[j+i+1];
            rep(k,N-i-j-2){
                int t1=Gi[j];
                int t2=Gi[j+k+1];
                int t3=Gj[k];

                if (t1+t2<t3 || t1+t3<t2 || t2+t3<t1){
                    can_make=false;
                    break;
                }
            }
        }
    }

    if (can_make){
        cout<<"Yes\n";
    }   
    else{
        cout<<"No\n";
    }

    return 0;
    
}
